frontend/models/product-list/sections/reusable-sections.ts

```diff
@@ -0,0 +1,143 @@
+import { filterFalsyItems } from '@helpers/application-helpers';
+import type { ComponentMiscPlacementFiltersInput } from '@lib/strapi-sdk';
+import { ProductListFieldsFragment } from '@lib/strapi-sdk';
+import { FAQ, faqFromStrapi } from '@models/components/faq';
+import type { ReusableSection } from '@models/reusable-section';
+import { findReusableSections } from '@models/reusable-section/server';
+import type { Dictionary } from 'lodash';
+import keyBy from 'lodash/keyBy';
+
+interface FindProductListReusableSectionsArgs {
+   strapiProductList: ProductListFieldsFragment | null | undefined;
+   ancestorHandles: string[];
+}
+
+export async function findProductListReusableSections({
+   strapiProductList,
+   ancestorHandles,
+}: FindProductListReusableSectionsArgs): Promise<ReusableSection[]> {
+   const reusableSections = await findReusableSections({
+      filters: {
+         placement: {
+            or: getPlacementConditions({
+               strapiProductList,
+               ancestorHandles,
+            }),
+         },
+      },
+   });
+
+   return reusableSections.map((reusableSection) => {
+      switch (reusableSection.section.type) {
+         case 'FAQs': {
+            const ancestryFaqsByCategory =
+               getAncestryFAQsByCategory(strapiProductList);
+            const ancestryFaqsWithoutCategory =
+               getAncestryFAQsWithoutCategory(strapiProductList);
```

Comment:
I didn't think about that 🤔 Seems unlikely to me as well, and even if we did I think it's unlikely that this would cause significant performance issue. I'd say let's deal with this if it ever becomes a problem